### PR TITLE
Update visualization docs

### PR DESCRIPTION
**Pull request by OwenArnold**
Part of #1478 also addresses #1559

Update visualisation documentation

### CODE REVIEW

File: docs/visualization/plotting-1d-data.ipynb [hunk 1]
```diff
@@ -128,7 +130,7 @@
    "source": [
     "### Overplotting using a dict of DataArrays\n",
     "\n",
-    "One can also supply the `plot` function with a `dict` of data arrays, and compatible data arrays will be overplotted on the same axes:"
+    "One can also supply the `plot` function with a `dict` of data arrays, which will then be overplotted on the same axes."
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-1d-data.ipynb:133]: Why the change? We had a case yesterday where mismatching units led to the use of different axes, so I think the original was more correct? Maybe just clarify the original if unclear?

File: docs/visualization/plotting-overview.ipynb [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "\n",
     "The difference between the two possible plot functions is that the free function can accept more input types than just the Scipp objects.\n",
     "It can also plot raw numpy arrays, as well as python dicts of Scipp variables or data arrays.\n",
+    "For Scipp object, the output plot will be the same with either approach: \n",
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-overview.ipynb:20]:
```suggestion
    "For Scipp objects, the produced plot will be the same with either approach: \n",
```

File: docs/visualization/plotting-overview.ipynb [hunk 3]
```diff
@@ -66,14 +67,25 @@
...
     "plot(d)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice in above that the dataset contains three data items.\n",
+    "The two 1D items with dimension y are combined into a single 1D plot.\n",
+    "The third item is 2D and plotted separately.\n",
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-overview.ipynb:82]:
```suggestion
    "The two 1-D items with dimension y are combined into a single 1-D plot.\n",
    "The third item is 2-D and plotted separately.\n",
```

File: docs/visualization/plotting-overview.ipynb [hunk 4]
```diff
@@ -66,14 +67,25 @@
...
+    "\n",
+    "All plots will use dimension coordinates values for each axis by default if they are available.\n",
+    "For the plot above, x and y dimensions have associated coordinates.\n",
+    "Notice the the units on the axis are also taken from the coordinates."
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-overview.ipynb:86]:
```suggestion
    "Notice that the units on the axes are also taken from the coordinates."
```

File: docs/visualization/plotting-overview.ipynb [hunk 5]
```diff
@@ -123,7 +135,14 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "plot(d['y', 4:7])"
+    "plot(d['y', 4 * sc.units.us : 7 * sc.units.us])"
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-overview.ipynb:138]: Any reason for using value-based indexing here? I feel it makes the plotting docs more confusing, since not every user will use value-based indexing (or have read that docs page already).

File: docs/visualization/plotting-overview.ipynb [hunk 6]
```diff
@@ -142,7 +161,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "plot(d['x', 4])"
+    "plot(d['x', 4 * sc.units.m])"
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-overview.ipynb:164]: Why the change (see above)?

File: docs/visualization/plotting-overview.ipynb [hunk 7]
```diff
@@ -187,6 +204,37 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "*Note*\n",
+    "    \n",
+    "Your plotting can be stored in a `config.yaml` file.\n",
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-overview.ipynb:233]: I don't understand this sentence.

**Review by SimonHeybrock — CHANGES_REQUESTED** [review]

File: docs/visualization/plotting-overview.ipynb (hunk 5 above)
Comment (OwenArnold) [reply docs/visualization/plotting-overview.ipynb:138]: Since the plots have sorted coordinates, it seemed like a good opportunity to work with coordinate values directly rather than indices. I think you are probably right about this though, it forces us to introduces/educate on sections that the reader might not be familiar with.

File: docs/visualization/plotting-nd-data.ipynb [hunk 8]
```diff
@@ -77,6 +75,18 @@
     "- Each dimension control comes with a `Continuous Update` checkbox, which is applied by default.\n",
     "If this is unselected, the plot will only update once the slider position has been released.\n",
     "\n",
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "*Note*\n",
+    "    \n",
+    "You can automatically recaluclate the intensity range using the `Rescale` button on the left of the plot.\n",
```
Comment (SimonHeybrock) [inline docs/visualization/plotting-nd-data.ipynb:82]:
```suggestion
    "You can automatically recalculate the intensity range using the `Rescale` button on the left of the plot.\n",
```

File: docs/visualization/plotting-1d-data.ipynb [hunk 9]
```diff
@@ -110,7 +110,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We can always plot just a single item of the dataset:"
+    "We can always plot just a single item of the dataset.\n",
+    "There is nothing special that needs to be done for plot itself to allow this.\n",
```
Comment (nvaytet) [inline docs/visualization/plotting-1d-data.ipynb:114]: I'm not sure how much information these 2 extra sentences add. I think the code example is rather self-explanatory?

File: docs/visualization/plotting-1d-data.ipynb [hunk 10]
```diff
@@ -464,6 +478,24 @@
...
     "plot(d4, filename='my_1d_figure.pdf')"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Help\n",
+    "\n",
+    "Inspect help to view all options for plot"
```
Comment (nvaytet) [inline docs/visualization/plotting-1d-data.ipynb:488]: I think this is expected of any well contructed python package, I don't think we need to spell it out in the docs

File: docs/visualization/plotting-2d-data.ipynb [hunk 11]
```diff
@@ -73,7 +73,7 @@
    "source": [
     "## Changing the colorscale\n",
     "\n",
-    "Changing the colorscale is handled in a similar way to the Matplotlib API. The colormap is defined by the `cmap` argument:"
+    "Changing the colorscale is handled in a similar way to [Matplotlib syntax](https://matplotlib.org/tutorials/colors/colormaps.html). The colormap is defined by the `cmap` argument:"
```
Comment (nvaytet) [inline docs/visualization/plotting-2d-data.ipynb:76]: New line?

File: docs/visualization/plotting-nd-data.ipynb [hunk 12]
```diff
@@ -68,9 +66,26 @@
     "\n",
     "- Each dimension comes with two sliders to control the position of the slice and its thickness.\n",
     "\n",
-    "- Upon figure creation, the thickness is set to cover the entire range of the additional dimension, and the data is summed along these dimensions.\n",
+    "- Upon figure creation, the thickness is set to zero for the additional dimension.\n",
```
Comment (nvaytet) [inline docs/visualization/plotting-nd-data.ipynb:69]: This is not striclty zero, it is selecting the first bin. So the thickness is whatever the width of that bin is (as reflected in the range of values displayed next to the slider).
Then, when we increase the thickness, it is selecting integer numbers of bins.

File: docs/visualization/plotting-nd-data.ipynb [hunk 13]
```diff
@@ -68,9 +66,26 @@
...
+    "Only data intersecting this 0-thickness plane is displayed.\n",
+    "The thickness can be increased and the and the data is summed along these dimensions between it's limits.\n",
+    "\n",
+    "- Changing the slice thickness will also change the color range, and the `Rescale` button can be used to automatically rescale the colorbar to the limits of the currently displayed data.\n",
+    "\n",
+    "- Each dimension control comes with a `Continuous Update` checkbox, which is applied by default.\n",
+    "If this is unselected, the plot will only update once the slider position has been released.\n",
```
Comment (nvaytet) [inline docs/visualization/plotting-nd-data.ipynb:76]:
```suggestion
    "If this is unselected, the plot will only update once the slider has been released.\n",
```

**Review by nvaytet — CHANGES_REQUESTED** [review]

File: docs/visualization/plotting-1d-data.ipynb (hunk 9 above)
Comment (OwenArnold) [reply docs/visualization/plotting-1d-data.ipynb:114]: I was trying to distinguish between plot options, and things that are core to scipp. I don't have a particularly strong preference, so will remove

File: docs/visualization/plotting-nd-data.ipynb [hunk 14]
```diff
@@ -68,9 +66,26 @@
     "\n",
     "- Each dimension comes with two sliders to control the position of the slice and its thickness.\n",
     "\n",
-    "- Upon figure creation, the thickness is set to cover the entire range of the additional dimension, and the data is summed along these dimensions.\n",
+    "- Upon figure creation, the thickness is set to the first bin width.\n",
+    "Only data intersecting this thickness plane is displayed.\n",
```
Comment (nvaytet) [inline docs/visualization/plotting-nd-data.ipynb:70]:
```suggestion
    "Only the data contained in that bin is displayed.\n",
```

File: docs/visualization/plotting-nd-data.ipynb [hunk 15]
```diff
@@ -68,9 +66,26 @@
...
+    "The thickness can be increased by an integer number of bins, and the and the data is operated on along these dimensions between its limits (either sum or mean, see note below).\n",
```
Comment (nvaytet) [inline docs/visualization/plotting-nd-data.ipynb:71]:
```suggestion
    "The thickness can be increased by an integer number of bins, and the data inside those bins will either be summed or averaged (see note below).\n",
```

**Review by nvaytet — CHANGES_REQUESTED** [review]

**Review by nvaytet — APPROVED** [review]

**Review by SimonHeybrock — APPROVED** [review]
